### ISSUE TITLE
Create testing script

### ISSUE THREAD

**Issue by greenmanspirit** [greenmanspirit/journal.txt#20]
After last bugs (https://github.com/greenmanspirit/journal.txt/issues/18 and https://github.com/greenmanspirit/journal.txt/issues/19) it has become apparent that the application has become complicated enough that hands on interactive testing is not enough and I will undoubtedly miss something.

**Comment by greenmanspirit** [comment]
I am making progress, I have created my own vim which is just a shell script that prints whatever I have in the environmental variable VALUE to the tmp file given. I am doing the testing in bash which probably isn't the most elegant way but it works for me since all I care about is the state of journal.txt after my program is run.

**Comment by greenmanspirit** [comment]
Oh and I found a bug while doing this, already proving useful

**Comment by greenmanspirit** [comment]
I have created an initial testing suite. 
test - This is a file containing all the the tests that need to be run
test_functions - This file contains all the utility functions I created for this
vim - This is a fake vim that writes whatever is in $VALUE to the given file

I will revisit these files later and comment them a bit better. It is late and I want this commited and pushed since it took a bit of thought to work out and I don't trust my laptop.
